src/JSInterop/Microsoft.JSInterop/src/DotNetObjectRefOfT.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.JSInterop
+{
+    /// <summary>
+    /// Wraps a JS interop argument, indicating that the value should not be serialized as JSON
+    /// but instead should be passed as a reference.
+    ///
+    /// To avoid leaking memory, the reference must later be disposed by JS code or by .NET code.
+    /// </summary>
+    /// <typeparam name="TValue">The type of the value to wrap.</typeparam>
+    public sealed class DotNetObjectRef<TValue> : IDotNetObjectRef, IDisposable where TValue : class
+    {
+        private long? _trackingId;
+
+        /// <summary>
+        /// This API is for meant for JSON deserialization and should not be used by user code.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public DotNetObjectRef()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="DotNetObjectRef{TValue}" />.
+        /// </summary>
+        /// <param name="value">The value to pass by reference.</param>
+        public DotNetObjectRef(TValue value)
```

Comment:
I know you might not want to change this around further, but might there be value in making this constructor `internal`?

That would force people to always use `DotNetObjectRef.Create<T>()`, so we know there's only one entrypoint to this whole system.

Reply:
👍